src/rgw/rgw_bucket_layout.h

```diff
@@ -122,10 +121,17 @@ void decode(bucket_log_layout& l, bufferlist::const_iterator& bl);
 void encode_json_impl(const char *name, const bucket_log_layout& l, ceph::Formatter *f);
 void decode_json_obj(bucket_log_layout& l, JSONObj *obj);
 
-
 struct bucket_log_layout_generation {
   uint64_t gen = 0;
   bucket_log_layout layout;
+  operator bucket_index_layout_generation() const {
+    bucket_index_layout_generation bilg;
+    bilg.gen = gen;
+    bilg.layout.type = BucketIndexType::Normal;
```

Comment:
* not all `bucket_log_layout_generation`s will be convertible to `bucket_index_layout_generation` - that's only the case for `BucketLogType::InIndex`
* `bucket_log_layout_generation::gen` is not necessarily the same as the index generation; we store that separately in `bucket_log_layout::in_index.gen`

so i feel like this conversion operator should be in `struct bucket_index_log_layout` instead so we can write code like this:
```c++
// bucket_log_layout_generation log;
if (log.layout.type == BucketLogType::InIndex) {
  bucket_index_layout_generation index = log.layout.in_index;
```
that or a free function taking `bucket_index_log_layout` like:
```c++
if (log.layout.type == BucketLogType::InIndex) {
  auto index = log_to_index_layout(log.layout.in_index);
```

Reply:
Fixed.